### DISCUSSION TITLE
nanobuild@v1 GitHub action fails due to 403 rate limit response

### DISCUSSION THREAD

**Discussion by CoryCharlton** [nanoframework/Home#1497] [General]
### Description

The nanobuild action occasionally fails due to GitHub rate limiting:

![image](https://github.com/nanoframework/Home/assets/10094287/b9118f9e-a2d2-4421-9dd2-19265646d61d)

### How to solve the problem

Change the URL used to locate and download the extension. Potentially use something like Cloudflare to cache the data.

**Comment by josesimoes** [comment]
This can be minimized if you pass your github token, instead of making an unauthenticated request. That can give you 5k request per hour.
Please see here:
https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28

With the current mechanism, which is querying github releases, we don't have any extra work. Mirroring this will incur in another tool/website to maintain...

**Reply by CoryCharlton** [reply]
Thanks for the info. For the "mirroring" I was thinking of a simple passthrough API with a short response cache time and allow CloudFlare to handle the caching.

Passing my token is a simpler solution. 

That being said I'll submit a PR to nanobuild to check for a 403 response + lack of token and output the link above so the solution is more visible to the next person that encounters this issue 👍

**Reply by josesimoes** [reply]
I see...
On the PR, OK if you think it's worth the effort. This is already mentioned on the readme, but I get it that it could be missed. 😁
